### DISCUSSION TITLE
How to pass context from DGS service to another via gateway

### DISCUSSION THREAD

**Discussion by saravind-hotstar** [Netflix/dgs-framework#712] [Q&A]
We want to pass enriched context from DGS service to another is there a standard way. Using @key @requires @external establishes relations but we would like to pass user context and additonal inferred context from service to another. 
- graphql service 1 -> populates user related additonal context
- graph service 2 -> would like to access it for optimization 

We would like to have this as standard feature, so that we are able to influence and provide hints to service1, service 2

**Comment by paulbakker** [comment]
This is what `@requires` is for in federation. It forces the gateway to first fetch that field, and pass it to the service that requires it.

Our gateway at Netflix doesn't do anything "smart" when it comes to context, aside from setting the correct headers to downstream calls for authz etc.
